src/client/datascience/raw-kernel/rawKernel.ts

```
@@ -1,527 +1,273 @@
-                inputOptions
-            );
-
-            // Send off our input reply no futures or promises
-            this.jmpConnection.sendMessage(inputReplyMessage);
-        }
-    }
-
-    // On dispose close down our connection and get rid of saved futures
-    public dispose(): void {
-        if (!this.isDisposed) {
-            if (this.jmpConnection) {
-                this.jmpConnection.dispose();
-            }
-
-            // Dispose of all our outstanding futures
-            this.futures.forEach((future) => {
-                future.dispose();
-            });
-            this.futures.clear();
-
-            this.isDisposed = true;
-        }
-    }
-    public shutdown(): Promise<void> {
-        throw new Error('Not yet implemented');
-    }
-    public getSpec(): Promise<Kernel.ISpecModel> {
-        throw new Error('Not yet implemented');
+    ): Kernel.IShellFuture<
+        KernelMessage.IShellMessage<T>,
+        KernelMessage.IShellMessage<KernelMessage.ShellMessageType>
+    > {
+        return this.realKernel.sendShellMessage(msg, expectReply, disposeOnDone);
     }
     public sendControlMessage<T extends KernelMessage.ControlMessageType>(
-        _msg: KernelMessage.IControlMessage<T>,
-        _expectReply?: boolean,
-        _disposeOnDone?: boolean
-    ): Kernel.IControlFuture<KernelMessage.IControlMessage<T>> {
-        throw new Error('Not yet implemented');
+        msg: KernelMessage.IControlMessage<T>,
+        expectReply?: boolean,
+        disposeOnDone?: boolean
+    ): Kernel.IControlFuture<
+        KernelMessage.IControlMessage<T>,
+        KernelMessage.IControlMessage<KernelMessage.ControlMessageType>
+    > {
+        return this.realKernel.sendControlMessage(msg, expectReply, disposeOnDone);
     }
     public reconnect(): Promise<void> {
-        throw new Error('Not yet implemented');
+        return this.realKernel.reconnect();
```

I think we should throw an exception here as this is not supported.